src/storage/src/source/generator.rs

```diff
@@ -88,6 +88,12 @@ impl SourceRender for LoadGeneratorSourceConnection {
 
         let (mut data_output, stream) = builder.new_output();
 
+        let primary_source_idx = config
+            .source_exports
+            .keys()
+            .position(|export_id| export_id == &config.id)
+            .expect("primary source must be included in exports");
+
```

Comment:
Is it worth making this a method on `RawSourceCreationConfig`? Seems that we do this in a few places